### PR TITLE
Improve error message

### PR DESCRIPTION
**Pull request by mrvoorhe**
Include the name of the method in the exception that is thrown when there is an exception marking a method

### CODE REVIEW

File: linker/Mono.Linker.Steps/MarkStep.cs [hunk 1]
```diff
@@ -153,7 +153,11 @@ void ProcessQueue ()
 			while (!QueueIsEmpty ()) {
 				MethodDefinition method = (MethodDefinition) _methods.Dequeue ();
 				Annotations.Push (method);
-				ProcessMethod (method);
+				try {
+					ProcessMethod (method);
+				} catch (Exception e) {
+					throw new Exception (string.Format ("Error processing method: '{0}' in assembly: '{1}'", method.FullName, method.Module.Name), e);
```
Comment (marek-safar) [inline linker/Mono.Linker.Steps/MarkStep.cs]: This should be different Exception than just Exception

Comment (mrvoorhe) [inline linker/Mono.Linker.Steps/MarkStep.cs]: @marek-safar Do you know if there is an existing exception type that would make sense to use?  Or should I introduce a "FatalLinkerException" ?

Comment (marek-safar) [inline linker/Mono.Linker.Steps/MarkStep.cs]: I think there is no existing exception you could use. What about `MarkerException` ?

File: linker/Mono.Linker.Steps/MarkStep.cs [hunk 2]
```diff
@@ -153,7 +153,11 @@ void ProcessQueue ()
...
+				}
 				Annotations.Pop ();
```
Comment (marek-safar) [inline linker/Mono.Linker.Steps/MarkStep.cs]: This should go to `finally`

**Comment by mrvoorhe** [comment]
@marek-safar Pop moved into finally and now throwing MarkException

File: linker/Mono.Linker.Steps/MarkStep.cs [hunk 3]
```diff
@@ -153,8 +153,15 @@ void ProcessQueue ()
 			while (!QueueIsEmpty ()) {
 				MethodDefinition method = (MethodDefinition) _methods.Dequeue ();
 				Annotations.Push (method);
-				ProcessMethod (method);
-				Annotations.Pop ();
+				try {
```
Comment (marek-safar) [inline linker/Mono.Linker.Steps/MarkStep.cs]: You don't need double try, do try/catch/finally instead

**Pull request merged**